internal/langserver/langserver.go

```diff
@@ -105,6 +106,10 @@ func (ls *langServer) StartAndWait(reader io.Reader, writer io.WriteCloser) erro
 		cancelFunc()
 	}()
 
+	ls.SetLogger(logging.NewLspLogger(&logging.LspLogger{
+		Context: ctx,
```

Comment:
In general storing context is considered anti-pattern in Go. We could pass the server instance here instead of the context. It would also make the contract much clearer, where the `LspLogger` depends on "LSP server" rather than arbitrary context which may or may not contain the instance of the server.